src/compiler/program.ts

```diff
@@ -2327,6 +2330,9 @@ namespace ts {
                     // only through top - level external module names. Relative external module names are not permitted.
                     if (moduleNameExpr && isStringLiteral(moduleNameExpr) && moduleNameExpr.text && (!inAmbientModule || !isExternalModuleNameRelative(moduleNameExpr.text))) {
                         imports = append(imports, moduleNameExpr);
+                        if (!usesUriStyleNodeCoreModules && currentNodeModulesDepth === 0 && !file.isDeclarationFile) {
+                            usesUriStyleNodeCoreModules = startsWith(moduleNameExpr.text, "node:");

```

Comment:
It seems kind of unimportant here whether what follows `node:` is actually something we recognize in our list, particularly as this PR shows we have neglected to keep the list fully up to date. I do use the list in the single-file analysis so we know we can _stop_ looking at more module specifiers when we see an unprefixed one like `fs` or `path`.